numpy/core/src/multiarray/compiled_base.c

```diff
@@ -795,7 +795,52 @@ static int sequence_to_arrays(PyObject *seq,
     return 0;
 }
 
-/* Inner loop for unravel_index */
+/*
+ * Force-cast an array into intp if it is empty. Fail otherwise.
+ *
+ * op        - Pointer to the array pointer. Will be modified if
+ *             recasting is needed.
+ * paramname - Name to be used for error string.
+ */
+static int _cast_to_intp_or_fail(PyArrayObject **op,
```

Comment:
I was wondering about that and I wasn't sure which way it should be. I went with how `sequence_to_arrays()` did it since it was the nearest function. I will change this one to the "own line" style.